Sources/PointFree/Episode/Episode.swift

```diff
@@ -0,0 +1,212 @@
+import Css
+import Either
+import Foundation
+import Html
+import HtmlCssSupport
+import HttpPipeline
+import HttpPipelineHtmlSupport
+import Optics
+import Prelude
+import Styleguide
+
+let episodeResponse: Middleware<StatusLineOpen, ResponseEnded, Either<String, Int>, Data> =
+  fetchEpisode
+    >-> responseEpisode
+
+private func fetchEpisode(_ conn: Conn<StatusLineOpen, Either<String, Int>>) -> IO<Conn<HeadersOpen, Episode?>> {
+
+  let possibleEpisode = episodes.first(where: {
+    conn.data.left == .some($0.slug)
+      || conn.data.right == .some($0.id)
+  })
+
+  return conn.map(const(possibleEpisode))
+    |> writeStatus(possibleEpisode == nil ? .notFound : .ok)
+}
+
+private func responseEpisode(_ conn: Conn<HeadersOpen, Episode?>) -> IO<Conn<ResponseEnded, Data>> {
+
+  switch conn.data {
+  case .none:
+    return conn.map(const(unit))
+      |> respond(notFoundView)
+  case let .some(ep):
+    return conn.map(const(ep))
+      |> respond(episodeView.map(addHighlightJs >>> addGoogleAnalytics))
```

Comment:
right now views can opt into adding Highlight.js and google analytics by `map`ing with these lil helper functions. I like this a lot and think we should build more off of it.